### PR TITLE
[ampl-asl] Update to 20240201 and disable generated header when cross-compiling

### PR DESCRIPTION
**Pull request by WangWeiLin-MV**
Related https://github.com/microsoft/vcpkg/pull/40503

* Disable generating header file when cross compiling, use the same file as native compile
* Update to `2024-02-01` https://github.com/ampl/asl/commit/2f5d9de248c53a3063bba23af2013cd3db768bf8
* Remove upstream merged patches
* Clean and set license to `BSD-3-Clause`
* Support with `!uwp` for `src/solvers/func_add.c(436): error C2065: '_pclose': undeclared identifier` since the function not available to UWP apps
* Change `ampl-mp` dependency `unofficial-asl` to `ampl-asl`

### Checklist

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [ ] ~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] When updating the upstream version, the `"port-version"` is reset (removed from `vcpkg.json`).
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### Test

The port usage tests pass with the following triplets:
* x64-windows
* x86-windows (With `CMAKE_CROSSCOMPILING`)

### CODE REVIEW

**Review by MonicaLiu0311 — DISMISSED** [review]

File: ports/ampl-mp/fix-dependency-asl.patch [hunk 1]
```diff
@@ -6,7 +6,7 @@ index 13f11cb..92089a3 100644
    endif()
  endif ()
  
-+find_package(unofficial-asl CONFIG REQUIRED)
++find_package(ampl-asl CONFIG REQUIRED)
```
Comment (BillyONeal) [inline ports/ampl-mp/fix-dependency-asl.patch:9]: Does this need corresponding `find_dependency` changes?

**Comment by BillyONeal** [comment]
>Disable generating host-dependent header, use pre-provided header for cross compiling

I'm not seeing how this is host dependent. In fact, it's guarded by `CMAKE_CROSSCOMPILING` which suggests that upstream believes that it is *not* host dependent. Can you clarify the problem being fixed here?

**Comment by WangWeiLin-MV** [comment]
> > Disable generating host-dependent header, use pre-provided header for cross compiling
> 
> I'm not seeing how this is host dependent. In fact, it's guarded by `CMAKE_CROSSCOMPILING` which suggests that upstream believes that it is _not_ host dependent. Can you clarify the problem being fixed here?

Thanks for your checking, it's not related with host dependent, rather preventing using different header file when cross-compiling. I've corrected the comment.

File: ports/ampl-mp/fix-dependency-asl.patch (hunk 1 above)
Comment (WangWeiLin-MV) [reply ports/ampl-mp/fix-dependency-asl.patch:9]: Done. There was no dependency, but now it has been added.

**Review by MonicaLiu0311 — APPROVED** [review]

**Pull request merged**